src/qiboconnection/models/job_data.py

```diff
@@ -1,24 +1,49 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 """ Job Data Typing """
 # pylint: disable=too-many-instance-attributes
-from dataclasses import dataclass
+# pylint: disable=E1101
+from qibo.models import Circuit
+
+from qiboconnection.api_utils import deserialize_job_description, parse_job_responses_to_results
+from qiboconnection.models.job_response import JobResponse
+
+# pylint: disable=super-init-not-called
+
+
+class JobData(JobResponse):
+    """Data shown to the user when get_job() method is used. It includes job human-readable results and metadata."""
 
-from qibo.models.circuit import Circuit
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
 
-from qiboconnection.typings.enums import JobStatus, JobType
+        self.result = parse_job_responses_to_results(job_responses=[JobResponse.from_kwargs(**kwargs)])[0]
+        self.description = deserialize_job_description(base64_description=self.description, job_type=self.job_type)
 
+        if not isinstance(self.result, (dict, type(None))):
+            raise ValueError("Job result needs to be a dict!")
 
-@dataclass(slots=True)
-class JobData:
-    """Data shown to the user when get_job() method is used. It includes job human-readable results and
-    metadata.
-    """
+        if not isinstance(self.description, (dict, type(None), Circuit)):
+            raise ValueError("Job description needs to be a dict of a Qibo Circuit!")
```

Comment:
Is there a reason not to be calling `super().__init__()` on his innit? We are skipping the parent class atributes initialization. I think this is only working bc we are usually using the `from_kwargs()` constructor

Reply:
If I write super():__init__() just after the init then I get a TypeError when running the unit tests: TypeError: JobResponse.__init__() missing 9 required positional arguments: 'user_id', 'device...

Reply:
Alternatively I could do the same as in Runcard: explicitly type the necessary attributes. However, as JobData requires the same attributes as JobResponse, I thought inheriting from it was natural since then there was only a single source of truth regarding needed attributes